pkg/agent/route/route_windows.go

```diff
@@ -387,16 +519,79 @@ func (c *Client) DeleteSNATRule(mark uint32) error {
 
 // TODO: nodePortAddresses is not supported currently.
 func (c *Client) AddNodePort(nodePortAddresses []net.IP, port uint16, protocol binding.Protocol) error {
-	return util.ReplaceNetNatStaticMapping(antreaNatNodePort, "0.0.0.0", port, config.VirtualNodePortDNATIPv4.String(), port, string(protocol))
+	netNatStaticMapping := &util.NetNatStaticMapping{
+		Name:         antreaNatNodePort,
+		ExternalIP:   net.ParseIP("0.0.0.0"),
+		ExternalPort: port,
+		InternalIP:   config.VirtualNodePortDNATIPv4,
+		InternalPort: port,
+		Protocol:     protocol,
+	}
+	if err := util.ReplaceNetNatStaticMapping(netNatStaticMapping); err != nil {
+		return err
+	}
+	c.netNatStaticMappings.Store(fmt.Sprintf("%d-%s", port, protocol), netNatStaticMapping)
+	klog.V(4).InfoS("Added NodePort NetNatStaticMapping", "NetNatStaticMapping", netNatStaticMapping)
+	return nil
 }
 
 func (c *Client) DeleteNodePort(nodePortAddresses []net.IP, port uint16, protocol binding.Protocol) error {
-	return util.RemoveNetNatStaticMapping(antreaNatNodePort, "0.0.0.0", port, string(protocol))
+	key := fmt.Sprintf("%d-%s", port, protocol)
+	obj, found := c.netNatStaticMappings.Load(key)
+	if !found {
+		klog.V(2).InfoS("Not found corresponding NetNatStaticMapping", "port", port, "protocol", protocol)
+		return nil
+	}
+	netNatStaticMapping := obj.(*util.NetNatStaticMapping)
+	if err := util.RemoveNetNatStaticMapping(netNatStaticMapping); err != nil {
+		return err
+	}
+	c.netNatStaticMappings.Delete(key)
+	klog.V(4).InfoS("Deleted NodePort NetNatStaticMapping", "NetNatStaticMapping", netNatStaticMapping)
+	return nil
+}
+
+// addLoadBalancerIngressIPRoute is used to add route which is used to route LoadBalancer ingress IP to Antrea gateway.
+func (c *Client) addLoadBalancerIngressIPRoute(svcIPStr string) error {
+	linkIndex := c.nodeConfig.GatewayConfig.LinkIndex
+	gw := config.VirtualServiceIPv4
+	metric := util.MetricHigh
+	_, svcIPNet, _ := net.ParseCIDR(svcIPStr)
+
+	route := generateRoute(svcIPNet, gw, linkIndex, metric)
+	if err := util.ReplaceNetRoute(route); err != nil {
+		return fmt.Errorf("failed to install route for LoadBalancer ingress IP %s: %w", svcIPStr, err)
+	}
+	klog.V(4).InfoS("Added LoadBalancer ingress IP route", "route", route)
+	c.serviceRoutes.Store(svcIPStr, route)
+
+	return nil
+}
+
+// deleteLoadBalancerIngressIPRoute is used to delete route which is used to route LoadBalancer ingress IP to Antrea gateway.
+func (c *Client) deleteLoadBalancerIngressIPRoute(svcIPStr string) error {
+	linkIndex := c.nodeConfig.GatewayConfig.LinkIndex
+	gw := config.VirtualServiceIPv4
+	metric := util.MetricHigh
+	_, svcIPNet, _ := net.ParseCIDR(svcIPStr)
+
+	route := generateRoute(svcIPNet, gw, linkIndex, metric)
```

Comment:
Why not use the item in the cache directly since it is already saved in `addLoadBalancerIngressIPRoute` ?

Reply:
Do you mean that we can read the route from the cache, then modify it, install the route and put it back to the cache?

Reply:
Done